test/fail/ok/suggest-label.tc.ok

```diff
@@ -0,0 +1,2 @@
+suggest-label.mo:2.9-2.11: type error [M0083], unbound label fo
+Did you mean foo?
```

Comment:
If the suggestion was wrong, does it make the error message more confusing? Also, AI tends to listen to the suggestions. If it's wrong, we may lead it to a wrong direction.

Reply:
Maybe we could show this by default and then include a compiler to flag for AI code generation (see [this comment](https://github.com/dfinity/motoko/pull/4720/files#r1783587695)). In general, this would make it so we can fine-tune what the AI sees without sacrificing readability for humans.

Reply:
It's a bit strange for a strongly typed language to suggest an identifier based on distance metric instead of types...If there is an AI mode for error message, maybe just listing the current in scope env is enough for AI to make a decision?

Reply:
Ocaml does the distance metric thing. That's where I stole the idea from.

I expect AI won't generate any mistyped identifiers, but I was hoping this would catch the error of using `Text.toLowercase`  instead of `Text.toLower`. (c.f. https://docs.google.com/spreadsheets/d/1osfLjjlqr43FtVzUkYrS_oBk5699jfMXB_XRZZ52sqM/edit?usp=sharing)

Reply:
I looked at filtering the suggestion to those that match on type, but the type checker doesn't have the expected type available when checking identifiers or projections - these are always done in inference mode. 

Moreover, in the common cases, applications, the type of the function is actually inferred.
For resolving function, knowing the type of the expected result would help a little,
but we don't have that info readily available.

Reply:
That spreadsheet shows the errors the AI encountered, but it doesn't show whether it can fix itself after feeding back the error messages. I wonder if the current error message (displaying all the functions in the module) is enough for AI to fix it.

I'm fine that we explore this idea, but not sure if a wrong suggestion can lead AI to a wrong path. A few corner cases I can think of:

* The code uses `fold`, do we suggest `foldLeft` or `foldRight`
* The code uses `Array.exists` or `Array.forall`, which doesn't exist in the module. Then any suggestion seems to be wrong?

Reply:
Maybe later we could (also) generate a set of identifiers that are in scope and then filter them so that they fit into the context. But for now I think this is just fine.